### DISCUSSION TITLE
Extra feature request ideas

### DISCUSSION THREAD

**Discussion by Sy256** [helge17/tuxguitar#56] [General]
Hello, I had these ideas, they're not high priority was just wondering if you guys could make this in some way, I know you're busy. 

~~- Hear notes when transposing them with -+ keys : This one's probably the easiest to do. Right now you can hear the note when you plug it in but you can't hear it when you transpose it. I have a method of starting with one chord and then copy/pasting the chord and transposing (increment/decrement semitone) the whole thing with cmd and + - keys. Works great but I would like to be able to hear the chord as if I was entering a note each time.~~ done, thank you!

- Chord to bass : To be able to select measures of chords and keep only the bottom notes. In some inversions that wont be the root note but I dont care, it's close enough and can transpose from there.

- Select measures and find/replace from one note number to another. Good for drums. Like...from measure 1 to 4, replace all instances of 44 with 46 (closed hi hat to open hi hat)

- Copy paste rhythm only: Like say you have a whole note chord and you want to make it all eighths instead, or some other tricky rhythm. Copy the rhythm ONLY of another measure and paste it to that chord to have that chord as all whatever copied rhythm.

- When moving beat set a marker for locking beat beyond that point : I move the beat (beat > move beat left/right) a lot in trial and error, I have it mapped to shortcut keys. It would be great if you could set a temporary lock marker where the beat won't be affected past that marker, so you can move the beat in certain sections only.

- ability to lock tracks from copy/paste section: So when you copy/paste sections and you have 'all tracks' selected, you can lock certain tracks and they won't be affected by the paste. Or just change the 'all tracks' to an expanded selection of checkboxes if that's easier. Good for if you already wrote out a distinct melody on one track but just need everything under it to keep repeating. Currently I get around this by cut/pasting the melody away from the repeat sections and then moving it back after pasting the sections.

- duplicate detector : if a section is named verse x for example, this will somehow show that there's no changes from section to section. Probably the hardest thing to do. Like if you have a track on verse 1, and it matches the 8 or 16 bars of verse 2 / 3 / 4 (whatever you compare against) perfectly, it'll say its a duplicate, if it doesn't then we'll know there's slight changes between them, and should go look at that

Thanks, again these are just ideas that'd be really cool, no rush and no worries if its not possible.

**Comment by guiv42** [comment]
As you mentioned, first item of this list was clearly the easiest one. It's already been committed (fed4ab7) and will be available in next beta (right now if you build from source).
To be honest, all the other suggestions seem a bit too complex for my modest development skills. So you might need some patience, until other contributors join this initiative.

edit: first feature is now available in [releases](https://github.com/helge17/tuxguitar/releases), thank you @helge17

**Reply by Sy256** [reply]
Thanks! :) I can brainstorm ways to get around some of this so you don't have to program it as detailed. For example would it be possible to create a way to select EVERYTHING after the cursor in one press/key shortcut? That'd take care of the 'locking beat' request because I could then just easily copy/paste all the parts back after shifting the beat in the measures I want. So basically it'd work like 'extend selection right' which is already an existing command, just x100 or something.

**Reply by guiv42** [reply]
TuxGuitar is for sure missing a serious update of documentation: the feature you are suggesting already exists. Just associate a shortcut (menu Tools/Shortcuts) to action "Extend Selection to Last Measure"
![Screenshot_20230711_211529](https://github.com/helge17/tuxguitar/assets/129443524/4c1e2851-d33c-4aab-9905-c0a7c1b4ac3f)

**Reply by Sy256** [reply]
Oh perfect thank you!

**Reply by helge17** [reply]
> TuxGuitar is for sure missing a serious update of documentation

FYI, I'm currently updating the documentation, but this really takes some time...

**Comment by Sy256** [comment]
I have figured out you can export the song as a midi and then turn it into a CSV file, make changes like find/replace in there and turn it back into a midi, but if I'm going to do all that then I might as well just make the changes inside my DAW, reaper, instead and export them back to tuxguitar. Still, figured maybe making you guys aware of this capability you might have a way to implement it inside tuxguitar or something for the far future to do those more advanced things. https://www.fourmilab.ch/webtools/midicsv/

Also I would close and split this topic into individual FRs but they're last priority and hard, and I have suitable workarounds for now, so I just wanted to update you guys on that midicsv tool and keep this open for the future.

**Comment by blindfaced** [comment]
hi

an idea is to add Slides from Unspecified Pitches (higher or lower) to a specified note and Slides to Unspecified Pitches from a note to high or low unspecified pitch
and a more convenient drum notation tool GP like maybe
we  are on the way to emphasize TG
